components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/dao/RoleDAOImpl.java

```diff
@@ -653,16 +653,17 @@ public void updateIdpGroupListOfRole(String roleId, List<IdpGroup> newGroupList,
     public void deleteRole(String roleId, String tenantDomain) throws IdentityRoleManagementException {
 
         String roleName = getRoleNameByID(roleId, tenantDomain);
-        if (systemRoles.contains(roleName)) {
+        if (systemRoles.contains(roleName) && !isSubOrgByTenant(tenantDomain)) {
```

Comment:
Since isSubOrgByTenant method is private shall we rename it to `isOrganization` ?

Reply:
Resolved by https://github.com/wso2/carbon-identity-framework/pull/5694/commits/5d20ec1ebe5617f9fbeefc606d7e01a1d6d2185d